infrastructure/docker-compose.deploy.yml

```diff
@@ -1017,27 +1035,33 @@ secrets:
     external: true
 configs:
   openhim-console-conf.{{ts}}:
-    file: ./infrastructure/openhim-console-config.deploy.json
+    file: /opt/opencrvs/infrastructure/openhim-console-config.deploy.json
   hearth-check-dupe-plugin.{{ts}}:
-    file: ./infrastructure/hearth-plugins/checkDuplicateTask.js
+    file: /opt/opencrvs/infrastructure/hearth-plugins/checkDuplicateTask.js
   hearth-queryparam-extensions-conf.{{ts}}:
-    file: ./infrastructure/hearth-queryparam-extensions.json
+    file: /opt/opencrvs/infrastructure/hearth-queryparam-extensions.json
   metricbeat-conf.{{ts}}:
-    file: ./infrastructure/monitoring/beats/metricbeat.yml
+    file: /opt/opencrvs/infrastructure/monitoring/beats/metricbeat.yml
+  filebeat.{{ts}}:
+    file: /opt/opencrvs/infrastructure/monitoring/filebeat/filebeat.yml
   mongo-on-deploy.{{ts}}:
-    file: ./infrastructure/mongodb/on-deploy.sh
-  metricbeat-rollover-policy.{{ts}}:
-    file: ./infrastructure/monitoring/beats/rollover-policy.json
+    file: /opt/opencrvs/infrastructure/mongodb/on-deploy.sh
+  beat-rollover-policy.{{ts}}:
+    file: /opt/opencrvs/infrastructure/monitoring/beats/rollover-policy.json
   kibana.{{ts}}:
-    file: ./infrastructure/monitoring/kibana/kibana.yml
+    file: /opt/opencrvs/infrastructure/monitoring/kibana/kibana.yml
   apm.{{ts}}:
-    file: ./infrastructure/monitoring/apm/apm-server.yml
+    file: /opt/opencrvs/infrastructure/monitoring/apm/apm-server.yml
   logstash.{{ts}}:
-    file: ./infrastructure/monitoring/logstash/logstash.yml
+    file: /opt/opencrvs/infrastructure/monitoring/logstash/logstash.yml
   logstash-pipeline.{{ts}}:
-    file: ./infrastructure/monitoring/logstash/logstash.conf
+    file: /opt/opencrvs/infrastructure/monitoring/logstash/logstash.conf
   influxdb-conf.{{ts}}:
-    file: ./infrastructure/influxdb.conf
+    file: /opt/opencrvs/infrastructure/influxdb.conf
+  jvm-options.{{ts}}:
```

Comment:
What benefit is there to creating these config files with the deployment timestamp?

Reply:
For us, no direct benefit really. Docker swarm treats both configurations and secrets as immutable. It throws an error if you try to deploy a changed configuration file with the same name. I imagine to be able to do that, you'd have to shut down all containers before deployment so that no container or replica is referring to the old version of the configuration anymore. 

I imagine there are some indirect benefits (similar to immutability in programming) when we deploy and less obscure inconsistency errors. In the end, each deployment is fairly complex with hundreds of components.